arches_lingo/src/arches_lingo/components/scheme/SchemeLabel/components/SchemeLabelEditor.vue

```diff
@@ -1,99 +1,174 @@
 <script setup lang="ts">
-import { useGettext } from "vue3-gettext";
+import { inject, useTemplateRef, watch } from "vue";
 
+import { useRouter } from "vue-router";
+import { Form } from "@primevue/forms";
+
+import DateWidget from "@/arches_component_lab/widgets/DateWidget/DateWidget.vue";
 import NonLocalizedStringWidget from "@/arches_component_lab/widgets/NonLocalizedStringWidget/NonLocalizedStringWidget.vue";
 import ReferenceSelectWidget from "@/arches_controlled_lists/widgets/ReferenceSelectWidget/ReferenceSelectWidget.vue";
 import ResourceInstanceMultiSelectWidget from "@/arches_component_lab/widgets/ResourceInstanceMultiSelectWidget/ResourceInstanceMultiSelectWidget.vue";
-import DateWidget from "@/arches_component_lab/widgets/DateWidget/DateWidget.vue";
 
+import { createScheme, upsertLingoTile } from "@/arches_lingo/api.ts";
 import { EDIT } from "@/arches_lingo/constants.ts";
 
+import type { Component, Ref } from "vue";
+import type { FormSubmitEvent } from "@primevue/forms";
 import type { AppellativeStatus } from "@/arches_lingo/types.ts";
 
-const { $gettext } = useGettext();
-
 const props = defineProps<{
-    schemeLabel: AppellativeStatus | undefined;
+    tileData: AppellativeStatus | undefined;
+    componentName: string;
+    sectionTitle: string;
+    graphSlug: string;
+    nodegroupAlias: string;
+    resourceInstanceId: string | undefined;
+    tileId?: string;
 }>();
+
+const router = useRouter();
+
+const schemeEditorFormRef = inject<Ref<Component | null>>(
+    "schemeEditorFormRef",
+);
+
+const openEditor =
+    inject<(componentName: string, tileid?: string) => void>("openEditor");
+const refreshReportSection = inject<(componentName: string) => void>(
+    "refreshReportSection",
+);
+
+const formRef = useTemplateRef("form");
+
+watch(
+    () => formRef.value,
+    (formComponent) => (schemeEditorFormRef!.value = formComponent),
+);
+
+async function save(e: FormSubmitEvent) {
+    try {
+        const formData = Object.fromEntries(
+            Object.entries(e.states).map(([key, state]) => [key, state.value]),
+        );
+
+        if (!props.resourceInstanceId) {
+            const updated = await createScheme({
+                [props.nodegroupAlias]: [formData],
+            });
+
+            await router.push({
+                name: props.graphSlug,
+                params: { id: updated.resourceinstanceid },
+            });
+
+            openEditor!(
+                props.componentName,
+                updated[props.nodegroupAlias][0].tileid,
+            );
+        } else {
+            await upsertLingoTile(
```

Comment:
For a tile create and subsequent edit, new tiles are still being created (requests are going out without the newly minted tile id). On create, I think we need to inform the editor pane about the newly minted tile id, or else close the editor and force users to click the pencil.

Reply:
nice catch! There's a followup issue with the serializer for `SchemeLicense` i think #241